cloud/deploys/bin/azure/lib.sh

```diff
@@ -0,0 +1,65 @@
+#! /usr/bin/env bash
+
+# CHARSET is a regex pattern that matches the acceptable characters to 
+# use when generating a secret value
+readonly CHARSET='A-Za-z0-9!"#$%&'\''()*+,-./:;<=>?@[\]^_`{|}~'
+
+#######################################
+# Create resource group
+# Arguments:
+#   1: The resource group name 
+#   2: The location of the resource group
+#######################################
+function key_vault::create_resource_group() {
+    if [ $(az group exists --name "${1}") = false ]; then
+        az group create --name "${1}" -location "${2}"
+    fi
+}
+
+#######################################
+# Create key vault
+# Arguments:
+#   1: The resource group name for the key vault
+#   2: The region (e.g. EastUS) to create the key vault in
+#   3: The name of the key vault 
+#######################################
+function key_vault::create_vault() {
+    az keyvault create \
+        --name "${3}" \
+        --resource-group "${1}"\
+        --location "${2}" \
+        --enable-rbac-authorization
+}
+
+#######################################
+# Add a secret to the key vault
+# Arguments:
+#   1: The name of the key vault
+#   2: The name of the secret (used to identify it e.g. "postgres-password")
+#   3: The value of the secret
+#######################################
+function key_vault::add_secret() {
+    az keyvault secret set \
+        --vault-name "${1}" \
+        --name "${2}" \
+        --value "${3}"
+}
+
+#######################################
+# For each argument after the first, generates a secret value and adds it to the
+# key vault, using the argument as the secret name.
+# Arguments:
+#   1: The name of the key vault
+#   2..n: Names of the secrets to be created (e.g. "postgres-password")
+#######################################
+function key_vault::add_generated_secrets() {
+    local vault_name="${1}"
+    shift;
+    for key in "$@";
+    do
+        echo "Generating secret: ${key}"
+        secret_value="$(head /dev/urandom | tr -dc "${charset}" | cut -c -40)"
```

Comment:
not super sure about the local/shift/for loop dynamics but is this something we'd also want to be local?

Reply:
hmm +1 I'm pretty sure